app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionFragment.kt

```diff
@@ -174,24 +181,29 @@ class DataCollectionFragment : AbstractFragment(), BackPressListener {
     // Reset progress bar
     progressBar.max = (taskPosition.sequenceSize - 1) * PROGRESS_SCALE
 
+    val target = taskPosition.relativeIndex * PROGRESS_SCALE
     if (shouldAnimate) {
       progressBar.clearAnimation()
-      with(ValueAnimator.ofInt(progressBar.progress, taskPosition.relativeIndex * PROGRESS_SCALE)) {
-        duration = 400L
-        interpolator = FastOutSlowInInterpolator()
-        addUpdateListener { progressBar.progress = it.animatedValue as Int }
-        start()
-      }
+      ValueAnimator.ofInt(progressBar.progress, target)
+        .apply {
+          duration = 400L
+          interpolator = FastOutSlowInInterpolator()
+          addUpdateListener { progressBar.progress = it.animatedValue as Int }
+        }
+        .start()
     } else {
-      progressBar.progress = taskPosition.relativeIndex
+      progressBar.progress = target
     }
   }
 
   override fun onBack(): Boolean {
-    if (viewModel.uiState.value == UiState.TaskSubmitted) {
+    if (viewModel.uiState.value == DataCollectionUiState.TaskSubmitted) {
       // Pressing back button after submitting task should navigate back to home screen.
       navigateBack()
-    } else if (viewPager.currentItem == 0) {
+      return true
+    }
+
+    if (viewModel.isAtFirstTask()) {
```

Comment:
Why is this change needed?

Reply:
earlier it was managed through viewpager and multiple sources are there include sequence handler, now everything is run using sequence hence required.